src/pages/questionPage.js

```diff
@@ -1,38 +1,72 @@
 'use strict';
 
 import {
-  ANSWERS_LIST_ID,
-  NEXT_QUESTION_BUTTON_ID,
-  USER_INTERFACE_ID,
+    ANSWERS_LIST_ID,
+    NEXT_QUESTION_BUTTON_ID,
+    USER_INTERFACE_ID,
 } from '../constants.js';
 import { createQuestionElement } from '../views/questionView.js';
 import { createAnswerElement } from '../views/answerView.js';
 import { quizData } from '../data.js';
 
-export const initQuestionPage = () => {
-  const userInterface = document.getElementById(USER_INTERFACE_ID);
-  userInterface.innerHTML = '';
+export const initQuestionPage = (userInterface) => {
 
-  const currentQuestion = quizData.questions[quizData.currentQuestionIndex];
+    userInterface = document.getElementById(USER_INTERFACE_ID);
+    userInterface.innerHTML = '';
 
-  const questionElement = createQuestionElement(currentQuestion.text);
+    const currentQuestion = quizData.questions[quizData.currentQuestionIndex];
 
-  userInterface.appendChild(questionElement);
+    const questionElement = createQuestionElement(currentQuestion.text);
 
-  const answersListElement = document.getElementById(ANSWERS_LIST_ID);
+    userInterface.appendChild(questionElement);
+    let score = 0;
 
-  for (const [key, answerText] of Object.entries(currentQuestion.answers)) {
-    const answerElement = createAnswerElement(key, answerText);
-    answersListElement.appendChild(answerElement);
-  }
+    const selectAnswer = (e) => {
+        currentQuestion.selected = e.target.dataset.key;
 
-  document
-    .getElementById(NEXT_QUESTION_BUTTON_ID)
-    .addEventListener('click', nextQuestion);
-};
+        if (currentQuestion.selected === currentQuestion.correct) {
+            e.target.classList.add('Correct');
+            score++;
+            e.target.style.backgroundColor = 'green';
 
-const nextQuestion = () => {
-  quizData.currentQuestionIndex = quizData.currentQuestionIndex + 1;
+        } else {
+            e.target.classList.add('Wrong');
+            score;
+            e.target.style.backgroundColor = 'red';
+        };
+
+
+
+        score;
+    };
+
+
+
+    const answersListElement = document.getElementById(ANSWERS_LIST_ID);
+
+    for (const [key, answerText] of Object.entries(currentQuestion.answers)) {
+        const answerElement = createAnswerElement(key, answerText);
+        answerElement.setAttribute('data-key', key)
+        answersListElement.appendChild(answerElement);
+    }
+
+    for (const selection of answersListElement.children) {
+        selection.addEventListener('click', selectAnswer);
```

Comment:
We can move this line into the previous for loop. I would also suggest a change to make the code cleaner: Instead of having one handler function `selectAnswer`, I would have two functions, `correctAnswer` and `incorrectAnswer`. Then you can set the appropriate handler on the answerElement here, based on the `correct` field of `currentQuestion`.

Reply:
I moved it, but I have question about split the functions. I can ask it at 05:00pm